tst/test_tuner.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import logging
+from pathlib import Path
+
+import pytest
+
+from syne_tune import Tuner
+from syne_tune.backend.trial_status import Status
+from syne_tune.config_space import randint
+from syne_tune.optimizer.baselines import RandomSearch
+from syne_tune.util import script_height_example_path
+from tst.util_test import temporary_local_backend
+
+_parameterizations = [
+    ("dummy", "dummy", False, Status.stopped),  # Worker should be stopped after 0.5 second
+    ("dummy", "dummy", True, Status.completed),  # Worker should complete (NOT be stopped) after 0.5 second
+    ("dummy", "dummy", True, Status.completed),  # Worker should complete (NOT be stopped) after 0.5 second
+]
+
+
+@pytest.mark.parametrize("dummy, dummy2, wait_for_completion, desired_status", _parameterizations)
+def test_tuner_wait_trial_completion_when_stopping(dummy, dummy2, wait_for_completion, desired_status):
```

Comment:
Strnagely enough any changes to this parametrization results in a failure of `test_pause_and_resume()` in https://github.com/awslabs/syne-tune/blob/main/tst/blackbox_repository/test_simulated_backend.py#L36 . Investigation is pending.